File: src/main/java/com/ngolajr/ispmedia/entities/Notificacao.java
package com.ngolajr.ispmedia.entities;

import com.ngolajr.ispmedia.entities.enums.EstadoEntrega;
import com.ngolajr.ispmedia.entities.enums.TipoNotificacao;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class Notificacao {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    @ManyToOne
    private Utilizador emissor;
    @ManyToOne
    private Utilizador destinatario;
    private String descricao;
    private EstadoEntrega estadoEntregaNotificacao;
    private TipoNotificacao tipoNotificacao;
    @ManyToOne
    private GrupoConvite convite;
}

File: ISPMEDIA-ANGULAR/src/app/entities/Notificacao.ts
import {User} from "./User";
import {EstadoEntrega} from "./enums/EstadoEntrega";
import {TipoNotificacao} from "./enums/TipoNotificacao";
import {GrupoConvite} from "./GrupoConvite";

export class Notificacao {
  id!: number;
  emissor: User = new User;
  destinatario: User = new User();
  descricao: string="";
  estadoEntregaNotificacao: EstadoEntrega = EstadoEntrega.PENDENTE;
  tipoNotificacao!:TipoNotificacao;
  convite:GrupoConvite=new GrupoConvite();

  constructor() {
  }
}
